libpod/runtime_ctr.go

```diff
@@ -915,6 +915,44 @@ func (r *Runtime) evictContainer(ctx context.Context, idOrName string, removeVol
 	return id, cleanupErr
 }
 
+// DependMap populates of map of all containers depending on the specified container
+func (r *Runtime) DependMap(ctx context.Context, ctr *Container, ctrMap map[string]*Container) error {
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+	// Grab all containers
+	allctrs, err := r.GetAllContainers()
+	if err != nil {
+		return err
+	}
+	// Make sure current Container is mapped
+	if ctrMap[ctr.ID()] == nil {
+		ctrMap[ctr.ID()] = ctr
+	}
+	return r.dependMap(ctx, ctr, allctrs, ctrMap)
+}
+
+// dependMap populates of map of all containers depending on the specified container
+func (r *Runtime) dependMap(ctx context.Context, searchCtr *Container, allCtrs []*Container, ctrMap map[string]*Container) error {
+	for _, ctr := range allCtrs {
+		if ctr == searchCtr {
+			continue
+		}
+		if ctrMap[ctr.ID()] != nil {
+			continue
+		}
+		for _, cid := range ctr.Dependencies() {
```

Comment:
It should be safe to recurse onto the dependencies of the container, the DB has a built-in guarantee of no cycles.